### ISSUE TITLE
BUG: seccomp_export_pfc hangs on prioritized syscalls with no rules

### ISSUE THREAD

**Issue by varqox** [seccomp/libseccomp#117]
The program below hangs on seccomp_export_pfc() if the priority of a syscall was set, but no rule has been added for it.
```c
#include <stdio.h>
#include <seccomp.h>

int main() {
	scmp_filter_ctx ctx = seccomp_init(SCMP_ACT_KILL);
	if (ctx == NULL) {
		perror("seccomp_init()");
		return 1;
	}

	// Set the x86 architecture
	if (seccomp_arch_native() != SCMP_ARCH_X86) {
		seccomp_arch_add(ctx, SCMP_ARCH_X86);
		seccomp_arch_remove(ctx, SCMP_ARCH_NATIVE);
	}

	int rc = seccomp_syscall_priority(ctx, SCMP_SYS(mmap2), 255);
	// seccomp_rule_add(ctx, SCMP_ACT_ALLOW, SCMP_SYS(mmap2), 0);

	int stdout_fileno = 1;
	seccomp_export_pfc(ctx, stdout_fileno); // Hangs here

	return 0;
}
```

**Comment by drakenclimber** [comment]
Looks like the "if" statement at line [307 of gen_pfc.c](https://github.com/seccomp/libseccomp/blob/master/src/gen_pfc.c#L307) is the culprit.  The syscall is invalid, so we "continue", but we don't advance p_iter.

```
	while (p_iter != NULL) {
		if (!p_iter->sys->valid)
			continue;
		_gen_pfc_syscall(db->arch, p_iter->sys, fds);
		p_iter = p_iter->next;
	}
```

**Comment by pcmoore** [comment]
Thanks for the report @varqox, and I believe @drakenclimber is right about the root cause.  I'll queue this up for later in the week, but if someone wants to send me a PR all the better ;)

**Comment by drakenclimber** [comment]
@pcmoore - I'll try to get to this in the next couple days

**Comment by pcmoore** [comment]
Thanks @drakenclimber!  No worries if you can't get to it, just let me know.

**Comment by pcmoore** [comment]
Fixed with PR #131.
